### PR TITLE
Improve some interfaces about the `load` and `index`

### PR DESCRIPTION
**Pull request by SimFG**
#1160
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
@SimFG Please resolve the conflicts

**Comment by XuanYang-cn** [comment]
This PR changed `loading_progress`  API, which is incompatible with Milvus CI.

I'm going to hold this until @binbinlv helps to modify CI so no further PR of PyMilvus will be held by this PR.

/hold

File: pymilvus/client/grpc_handler.py [hunk 1]
```diff
@@ -743,91 +724,33 @@ def _check():
...
 
-            if ok:
-                return
+        def can_loop(t) -> bool:
+            return True if timeout is None else t <= (start + timeout)
 
+        while can_loop(time.time()):
+            progress = self.get_loading_progress(collection_name, partition_names, timeout=timeout)
+            if progress >= 100:
+                break
```
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:735]: the `break` should be `return`, or this function will raise exception when progress >= 100

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: pymilvus/client/grpc_handler.py (hunk 1 above)
Comment (SimFG) [reply pymilvus/client/grpc_handler.py]: Thanks!

**Review by XuanYang-cn — APPROVED** [review]

**Pull request merged**